library/authenticate/ftrack/library/authenticate/helper/credential.py

```diff
@@ -0,0 +1,137 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+
+import logging
+import keyring
+from abc import ABC, abstractmethod
+from typing import Optional, Dict
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+
+
+class CredentialProvider(ABC):
+    """
+    Abstract base class for credential providers.
+    """
+
+    @abstractmethod
+    def get_credential(self) -> Optional[Dict[str, str]]:
```

Comment:
Yes, lets talk about this, this is a good example where is much interesting to not use the properties because its a helper function exposed to the final users and methods gives them more feedback.

Reply:
Agreed to rename methods to store_credential and load_credential.